eth/bor_api_backend.go

```diff
@@ -48,35 +49,31 @@ func (b *EthAPIBackend) GetVoteOnRootHash(ctx context.Context, starBlockNr uint6
 		return false, errors.New("Only available in Bor engine")
 	}
 
-	localRootHash, err := api.GetRootHash(starBlockNr, endBlockNr)
-
+	localEndBlock, err := b.BlockByNumber(ctx, rpc.BlockNumber(endBlockNr))
 	if err != nil {
-		return false, err
+		return false, errEndBlock
 	}
 
-	if !b.eth.Miner().Mining() {
-		return localRootHash == rootHash, nil
+	localEndBlockHash := localEndBlock.Hash().String()
+
+	if err != nil {
+		return false, err
 	}
 
 	downloader := b.eth.handler.downloader
 	isLocked := downloader.LockMutex(endBlockNr)
 
 	if !isLocked {
 		downloader.UnlockMutex(false, "", common.Hash{})
-
-		return false, errors.New("Previous sprint is still in locked state")
+		return false, errors.New("Whitelisted number or locked sprint number is more than the received end block number")
 	}
 
-	if localRootHash != rootHash[2:] {
+	if localEndBlockHash != endBlockHash {
 		downloader.UnlockMutex(false, "", common.Hash{})
-
-		return false, errors.New("RootHash mismatch+" + localRootHash + rootHash)
+		return false, errors.New("EndBlockHash mismatch")
```

Comment:
could we have this error as a variable/constant + add error context with fmt.Errorf("%w: hash1 %s, hash2 %s", ...)?